### PR TITLE
More applicability edits

### PR DESCRIPTION
**Pull request by martinthomson**
Quite a few pieces of text in here that needed corrections.  As before, I'm happy to pull out individual commits so we can discuss.

### CODE REVIEW

File: draft-ietf-quic-applicability.md [hunk 1]
```diff
@@ -556,24 +561,35 @@ information is intended to be used by the network. QUIC has a long header that
...
-to forward the packets to that server without retaining connection state.
+order to support stateless load balancing.
+
+Server deployments with load balancers and other routing infrastructure need to
+ensure that this infrastructure consistently routes packets to the correct
+server instance. This might require coordination between servers and
+infrastructure. One method of achieving this involves encoding routing
+information into the connection ID. This ensures that there is no need to for
+servers and infrastructure to coordinate routing information for each
+connection. <!-- TODO quic-lb reference? -->
```
Comment (mirjak) [inline draft-ietf-quic-applicability.md:592]:
```suggestion
connection. See further {{QUIC-LB}}.
```

File: draft-ietf-quic-applicability.md [hunk 2]
```diff
@@ -664,8 +680,8 @@ will persist only for as long as any Version Negotiation packets take to be
 transmitted and responded to.  So the third stage can follow after a relatively
 short delay.
 
-The third stage completes the process by enabling validation of the negotiation
-version as though the new version were disabled.
+The third stage completes the process by enabling validation of the negotiated
+version with the assumption that the new version is fully available.
```
Comment (mirjak) [inline draft-ietf-quic-applicability.md:684]: Actually I'm not sure here. If I remember correctly this text was provide by someone else and it not entirely clear to me anymore what this stage is suppose to achieve....

**Review by mirjak — APPROVED** [review]

File: draft-ietf-quic-applicability.md [hunk 3]
```diff
@@ -419,78 +419,80 @@ being terminated in some protocols.
...
+Similarly, an application has usually no control about the length of a QUIC
+packet on the wire. QUIC provides the ability to add a PADDING frame to
+arbitrarily increase the size of packets.
+
+Padding is used by QUIC to ensure that the path is capable of transferring
+datagrams of at least a certain size, both during the handshake and for
+connection migration. Padding can also be used by an application to reduce
+leakage of information about the data that is sent. A QUIC implementation can
+expose an interface that allows an application layer to specify how to apply
+padding.
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:446]: Yeah, it was not precise enough for me and I don't think that it has much bearing on usage (it's more for implementation and so the core spec can deal with that).

File: draft-ietf-quic-applicability.md [hunk 4]
```diff
@@ -521,26 +524,28 @@ has been initiated by one endpoint (for a limited time period), the expectation
...
-and one for closures initiated by the application. An application using QUIC can
-define application-specific error codes (see, for example, {{QUIC-HTTP}},
-Section 8.1). In the case of a grateful shut-down initiated by the application
-after application layer negotiation, a NO_ERROR code is expected. Further,
-the CONNECTION_CLOSE frame provides an optional reason field, that can be used
+An immediate close will emit an CONNECTION_CLOSE frame. This frames has two
+sets of types: one for QUIC internal problems that might lead to connection
+closure, and one for closures initiated by the application. An application
+using QUIC can define application-specific error codes (see, for example,
+{{QUIC-HTTP}}, Section 8.1).
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:531]: The NO_ERROR thing is an HTTP error code, which isn't generic advice.  But more concretely, when you gracefully shutdown, you might agree to stay quite and run down the idle timeout instead.  Or there might be other ways of signaling clean shutdown prior to sending CONNECTION_CLOSE.  I just thought that this was too specific.

File: draft-ietf-quic-applicability.md [hunk 5]
```diff
@@ -556,24 +561,35 @@ information is intended to be used by the network. QUIC has a long header that
...
+Aside from the Destination Connection ID field of the first packets sent by
+clients, the connection ID can be zero length. This is a choice that is made by
+each endpoint individually.
+
+An endpoint that selects a zero-length connection ID will receive packets with
+a zero-length Destination Connection ID. The endpoint needs to use other
+information, such as its IP address and port number to identify which
+connection is referred to. An endpoint can choose to use the source IP address
+and port on datagrams, but this could mean that the endpoint is unable to match
+datagrams to connections successfully if these values change.
```
[the remark this answers is not in the document]
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:576]: How about:

```suggestion
datagrams to connections successfully if these values change, making migration
effectively impossible.
```

File: draft-ietf-quic-applicability.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:684]: This was my text originally.  But I got it backwards.  You can't complete an upgrade that adds a new version and end with the assumption that the new version is disabled.  I probably meant "enabled".

File: draft-ietf-quic-applicability.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mirjak) [reply draft-ietf-quic-applicability.md:531]: Ah, didn't realise that an HTTP errs code. Do we need to say more here?

File: draft-ietf-quic-applicability.md (hunk 2 above)
Comment (mirjak) [reply draft-ietf-quic-applicability.md:684]: Ah okay. However, what's actually meant by validation in this phase?

File: draft-ietf-quic-applicability.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:531]: I don't think so.  This is pretty well treated in the main spec and you have discussion on idle timeouts that seems ample.

File: draft-ietf-quic-applicability.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-quic-applicability.md:685]: Validation being authentication of the values.  Maybe I can use that instead.

File: draft-ietf-quic-applicability.md [hunk 6]
```diff
@@ -664,8 +681,8 @@ will persist only for as long as any Version Negotiation packets take to be
 transmitted and responded to.  So the third stage can follow after a relatively
 short delay.
 
-The third stage completes the process by enabling validation of the negotiation
-version as though the new version were disabled.
+The third stage completes the process by enabling validation of the negotiated
```
Comment (martinthomson) [inline draft-ietf-quic-applicability.md:684]:
```suggestion
The third stage completes the process by enabling authentication of the negotiated
```

**Pull request merged**